File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/ReverseTaxResults.js
/**
 * Structure returned when invoking the reverse tax calculation APIs.
 */

function ReverseTaxResults() {

    this.BaseSaleAmount = 0;
    this.Taxes = [];

    /**
     * Sets the base sale amount necessary to arrive at desired total tax.
     * @param baseSaleAmount Number
     * @return Reference to this object for method chaining. 
     */
    this.baseSaleAmount = function (baseSaleAmount) {
        this.BaseSaleAmount = baseSaleAmount;
        return this;
    }

    /**
     * Base sale amount necessary to arrive at desired total tax.
     * @return Number
     */
    this.getBaseSaleAmount = function () {
        return this.BaseSaleAmount;
    }

    /* Reference to this object for method chaining.*
     * Sets the base sale amount necessary to arrive at desired total tax.
     * @param baseSaleAmount Number 
     */
    this.setBaseSaleAmount = function (baseSaleAmount) {
        this.BaseSaleAmount = baseSaleAmount;
    }

    /**
     * Sets the taxes generated for the transaction.
     * @param taxes Array of TaxData
     * @return Reference to this object for method chaining. 
     */
    this.taxes = function (taxes) {
        this.Taxes = taxes;
        return this;
    }

    /**
     * Adds a TaxData item to this results object.
     * @param taxesItem TaxData object.
     * @return Reference to this object for method chaining. 
     */
    this.addTaxesItem = function (taxesItem) {
        this.Taxes.push(taxesItem);
        return this;
    }

    /**
     * Taxes generated for the transaction.
     * @return Array of TaxData
     */
    this.getTaxes = function () {
        return this.Taxes;
    }

    /* Reference to this object for method chaining.*
     * Sets the taxes generated for the transaction.
     * @param taxes Array of TaxData 
     */
    this.setTaxes = function (taxes) {
        this.Taxes = taxes;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/ReverseTaxResults.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

/**
 * Structure returned when invoking the reverse tax calculation APIs.
 */
public class ReverseTaxResults {

    @SerializedName("BaseSaleAmount")
    private Double baseSaleAmount = null;

    @SerializedName("Taxes")
    private List<TaxData> taxes = new ArrayList<>();

    /**
     * Sets the base sale amount necessary to arrive at desired total tax.
     * @param baseSaleAmount
     * @return 
     */
    public ReverseTaxResults baseSaleAmount(Double baseSaleAmount) {
        this.baseSaleAmount = baseSaleAmount;
        return this;
    }

    /**
     * Base sale amount necessary to arrive at desired total tax.
     * @return baseSaleAmount
     */
    public Double getBaseSaleAmount() {
        return baseSaleAmount;
    }

    /**
     * Sets the base sale amount necessary to arrive at desired total tax.
     * @param baseSaleAmount 
     */
    public void setBaseSaleAmount(Double baseSaleAmount) {
        this.baseSaleAmount = baseSaleAmount;
    }

    /**
     * Sets the taxes generated for the transaction.
     * @param taxes
     * @return 
     */
    public ReverseTaxResults taxes(List<TaxData> taxes) {
        this.taxes = taxes;
        return this;
    }

    /**
     * Adds a TaxData item to this results object.
     * @param taxesItem
     * @return 
     */
    public ReverseTaxResults addTaxesItem(TaxData taxesItem) {
        this.taxes.add(taxesItem);
        return this;
    }

    /**
     * Taxes generated for the transaction.
     * @return taxes
     */
    public List<TaxData> getTaxes() {
        return taxes;
    }

    /**
     * Sets the taxes generated for the transaction.
     * @param taxes 
     */
    public void setTaxes(List<TaxData> taxes) {
        this.taxes = taxes;
    }
}
